target/linux/ipq40xx/base-files/lib/upgrade/platform.sh

```diff
@@ -12,17 +146,15 @@ platform_do_upgrade() {
 	case "$(board_name)" in
 	asus,rt-acrh17|\
 	asus,rt-ac58u)
+		local magic=$(get_magic_long "$1")
 		CI_UBIPART="UBI_DEV"
 		CI_KERNPART="linux"
-
-		# support sysupgrade from initramfs boot
-		local ubidev=$(nand_find_ubi $CI_UBIPART)
-		local jffs2=$(nand_find_volume $ubidev jffs2)
-		local linux2=$(nand_find_volume $ubidev linux2)
-		[ -n "$jffs2" ] && ubirmvol /dev/$ubidev --name=jffs2
-		[ -n "$linux2" ] && ubirmvol /dev/$ubidev --name=linux2
-
-		nand_do_upgrade "$1"
+		if [ "$magic" == "27051956" ]; then
+			echo "Got Asus factory image."
+			asus_nand_upgrade_factory 50409472 "$1"
```

Comment:
@ptpt52 Be careful of the partition size here. I'm not sure if RT-AC58U used the same partition size as RT-ACRH17.